python/pyspark/sql/tests/test_dataframe.py

```diff
@@ -279,15 +282,6 @@ def test_drop_duplicates(self):
             messageParameters={"arg_name": "subset", "arg_type": "NoneType"},
         )
 
-        with self.assertRaises(PySparkTypeError) as pe:
-            df.dropDuplicates(None).show()
-
-        self.check_error(
-            exception=pe.exception,
-            errorClass="NOT_STR",
-            messageParameters={"arg_name": "subset", "arg_type": "NoneType"},
-        )
-
```

Comment:
@itholic Please let me know if deleting this test case sounds good to you... 

The way I made named parameter work is to redefine the parameter as:
```
def dropDuplicates(
        self, subset: Optional[Union[str, List[str]]] = None, *subset_varargs: str
    ) -> ParentDataFrame:
```

But this means that when "subset" is None, it can mean two things:
1. dropDuplicates(None)
2. dropDuplicates()

With my change it looks it's not possible to distinguish these two...

Reply:
Let's add it to the migration guide `python/docs/source/migration_guide/pyspark_upgrade.rst`. Also you can leverage `_NoValue` instance to distinguish `None` from not setting a value.

Reply:
Thank you! Changed to use _NoValue here